dataactbroker/fileRoutes.py

```diff
@@ -139,15 +141,15 @@ def sign_submission_file():
         return fileManager.get_signed_url_for_submission_file()
 
     @app.route("/v1/submission/<int:submission_id>/narrative", methods=['GET'])
-    @permissions_check(permission='reader')
-    def get_submission_narratives(submission_id):
-        return narratives_for_submission(int(submission_id))
+    @requires_submission_perms('reader')
+    def get_submission_narratives(submission):
+        return narratives_for_submission(submission)
 
     @app.route("/v1/submission/<int:submission_id>/narrative", methods=['POST'])
-    @permissions_check(permission='writer')
-    def post_submission_narratives(submission_id):
+    @requires_submission_perms('writer')
+    def post_submission_narratives(submission):
         json = request.json or {}
         # clean input
         json = {key.upper():value.strip() for key, value in json.items()
                 if isinstance(value, str) and value.strip()}
-        return update_narratives(int(submission_id), json)
+        return update_narratives(submission, json)
```

Comment:
Nah, we were wrapping that function with `@requires_submission_perms` before which did the conversion. We've just bumped that conversion into the routes now.